### PR TITLE
Adds dockerfile

### PR DESCRIPTION
**Pull request by tpickett**
I host all my smarthome stuff in a K8S lab cluster, and as such needed a docker image to run this with. I added a very simplistic docker file to aid in deploying to some container orchestration tool. 
I pushed my own image to dockerhub: https://hub.docker.com/repository/docker/tpickett/mimosa/general - but you may want to create your own image in dockerhub that is pushed to by github actions on releases. I can also help with that stuff as well if you want!

### CODE REVIEW

**Comment by zane797** [comment]
This is probably just my not understanding Docker properly, but doesn't this not expose the network ports? I had issues on my testing PC for a docker image for this without doing this. I could access it locally but could not connect to it from another PC on the network [Link](https://hub.docker.com/r/zane797/mimosa_inventory/tags)

**Comment by tpickett** [comment]
```EXPOSE 5000``` is what exposes the port.  i added it in the 2nd commit. but the expose directive is optional. publishing of ports is defined at runtime i.e. ```docker run```. 
for reference: https://www.devopsschool.com/blog/understanding-expose-in-dockerfile-with-example/#:~:text=Exposing%20ports%20is%20optional.,anywhere%2C%20even%20outside%20Docker%20host.

**Comment by FireMarshmellow** [comment]
There are a couple things I'd like to do with Github actions but I've never used them, I'd really appreciate your help if you're willing? Also I don't use Docker, but I'd definitely like to have the images out there.

**Comment by tpickett** [comment]
done. I added a spot to create your exe as that is the only thing that i saw that it looks like you are doing manually. I created a PR here: https://github.com/FireMarshmellow/M.I.M.O.S.A/pull/46

If you do merge that one in - it will auto merge this PR in as i had to cherry-pick these commits in for testing the docker image creation btw.

**Comment by Paillat-dev** [comment]
The port should not be published in the dockerfile because that would break more advanced usages with docker networks. Instead it should be exposed when the container is run. That can be facilitated with a docker compose file.

File: dockerfile [hunk 1]
```diff
@@ -7,4 +7,6 @@ RUN pip install --no-cache-dir -r requirements.txt
 
 COPY . .
 
+EXPOSE 5000
```
Comment (Paillat-dev) [inline dockerfile:10]: The port should probably not be exposed

**Comment by Paillat-dev** [comment]
> There are a couple things I'd like to do with Github actions but I've never used them, I'd really appreciate your help if you're willing? Also I don't use Docker, but I'd definitely like to have the images out there.

There is the *Publish Docker Container* actions workflow wich works out of the box, it just needs a Dockerfile. In it you can remove the cron build to only build on push on the main branch, then the image will autonatically be available in the github container registry.

**Comment by Paillat-dev** [comment]
Also a docker compose file could be added, as well as some documentation, for volumes notably for data persistence.

**Comment by pandaboy6621** [comment]
Is there a Mimosa v3 docker image anywhere? I got one working from @zane797 but it is only running as version 2.1.

**Comment by zane797** [comment]
> Is there a Mimosa v3 docker image anywhere? I got one working from @zane797 but it is only running as version 2.1.

Hey @pandaboy6621, there are two in distribution I think. One by tpickett and one by me. I just ran an update on mine which I believe should bring it to the latest and greatest. Let me know if it works for you, I'm going to get back to working on my deployment soon so it would be good to know if something about it is broken by the update.

Edit: I just updated the copy on my Unraid server. The main branch is v4 beta so the docker image is now v4 beta since my script pulls directly from the main branch files.

**Comment by pandaboy6621** [comment]
I recreated the container, pulled the image, and am now on the v4 beta. Thanks @zane797!

**Pull request closed without merging**